### ISSUE TITLE
Bug with interesting leap year pattern?

### ISSUE THREAD

**Issue by DABH** [harrisiirak/cron-parser#122]
I'm trying to parse a cron with the following pattern:
`0 2 29 2 0`
However, when I try, I get the error:
`Invalid explicit day of month definition`
This pattern should match all February 29th's that are on a Sunday (accounting for leap years in some reasonable way).  This is enough of an edge case that I figure it might actually be a bug.  Any ideas what's going on and how to fix this?

**Comment by harrisiirak** [comment]
@DABH thanks for reporting? Can i ask which version of the library you're using?
Latest version is currently producing results as such:

```
const parser = require('cron-parser');
const results = parser.parseExpression('0 2 29 2 0');

console.log(results.next().toDate()); // 2019-02-03T00:00:00.000Z
console.log(results.next().toDate()); // 2019-02-10T00:00:00.000Z
```

**Comment by harrisiirak** [comment]
@DABH +1

**Comment by fishcharlie** [comment]
I'm not getting any errors with @DABH's code he posted in the original issue. That being said, I'm getting this error in the latest version (2.6.0) of this package, using the code I have attached below.

```
const parser = require('cron-parser');
const results = parser.parseExpression('0 0 29 2 *');

console.log(results.next().toDate());
```

I'm running on Node.js version `v10.8.0`.

Below is the output I get when running the code above.

```
/Users/me/Desktop/tmp/node_modules/cron-parser/lib/expression.js:456
        throw new Error('Invalid explicit day of month definition');
        ^

Error: Invalid explicit day of month definition
    at CronExpression._findSchedule (/Users/me/Desktop/tmp/node_modules/cron-parser/lib/expression.js:456:15)
    at CronExpression.next (/Users/me/Desktop/tmp/node_modules/cron-parser/lib/expression.js:549:23)
    at Object.<anonymous> (/Users/me/Desktop/tmp/index.js:4:21)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:742:12)
    at startup (internal/bootstrap/node.js:266:19)
```

As you can see from [this link](https://crontab.guru/#0_0_29_2_*) the code above should print `2020-02-29T00:00:00.000Z`.

**Comment by fishcharlie** [comment]
I have created a PR for this fix (#135). Let me know if you have any questions!
